### ISSUE TITLE
Why do backpropagation for original VGG net?

### ISSUE THREAD

**Issue by fengyuleidian0615** [fzliu/style-transfer#45]
Hi
I've been reading the original paper and this design for a couple of days, while one implementation decision puzzled me.

By my understanding, the original VGG model from caffe is used to get content from the input image, and then turn content into style(Gram matrices ), the loss is defined by a weighted combination of feature loss and style loss. It appears original VGG models does not involve any back-prorogation stuff, may I ask is there any reason for the design about this part.

The relevant code is [here](https://github.com/dpaiton/style-transfer/blob/master/style.py#L241)

Thanks!

**Comment by fengyuleidian0615** [comment]
hmm, seems I misunderstood this part with back-propagation involved training part.
Back-propagation part is aimed to grab the gradient for the input images, and it's the [minimize](https://github.com/fzliu/style-transfer/blob/master/style.py#L462) role to change the input images w.r.t the gradient.
